Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -297,7 +336,18 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
     }
 
     func testArray() throws {
-        // Skipping until there is a BSON.init(fromExtJSON) because its recursive
+        // Success cases
+        expect(try Array(fromExtJSON: [1, ["$numberLong": "2"], "3"], keyPath: []))
+            .to(equal([BSON.int32(Int32(1)), BSON.int64(Int64(2)), BSON.string("3")]))
```

Comment:
the compiler will be able to infer that 1 and 2 here are `Int32` and `Int64` respectively by nature of being included in a `BSON.int32` invocation, so the casts can be omitted.

Reply:
ok